src/core/server/http/router/route.ts

```diff
@@ -146,10 +146,19 @@ export interface RouteConfigOptions<Method extends RouteMethod> {
   body?: Method extends 'get' | 'options' ? undefined : RouteConfigOptionsBody;
 
   /**
-   * Timeouts for processing durations. Response timeout is in milliseconds.
-   * Default value: 2 minutes
+   * Defines per-route timeouts.
    */
-  timeout?: number;
+  timeout?: {
+    /**
+     * Milliseconds to receive the payload
+     */
+    payload?: Method extends 'get' | 'options' ? undefined : number;
```

Comment:
Per https://github.com/elastic/kibana/pull/73730#discussion_r465002635, were you suggesting that we rename `payload` to `payloadParsing`? I  don't like just calling this the `payload` timeout, but I'm afraid of reusing "parsing" here as that might make users think it relates to https://github.com/elastic/kibana/blob/2c939ea61d580d23d5df606c8b795f9f91380558/src/core/server/http/router/route.ts#L110

Reply:
+1 to `payloadParsing` not being correct and maybe confusing since this timeout does not have to do with parsing. I actually prefer the current naming, though if I had my way, we'd stop using the word `payload` and always use `body` since that is what the HTTP spec calls it.